nabs/tests/test_plans.py

```diff
@@ -0,0 +1,207 @@
+    egu = 'm'
+
+
+class SimDelayStage(DelayBase):
+    motor = Cpt(SimDelayMotor)
+
+
+@pytest.fixture(scope='function')
+def time_motor():
+    return SimDelayStage('SIM', name='sim', egu='s', n_bounces=1)
+
+
+@pytest.mark.timeout(PLAN_TIMEOUT)
+def test_delay_scan(RE, time_motor):
+    """
+    Check the delay scan, verify that velo is set appropriately.
+    """
+    logger.debug('test_delay_scan')
+
+    # Speed of light is more or less 3e8
+    goal = 1/(3e8)
+    msgs = nbp.delay_scan(time_motor, [0, goal], 1, duration=0.01)
+    moves = list(msg.args[0] for msg in msgs if msg.command == 'set')
+    # first point is the velo, which should be close to 1 with 1 bounce set
+    assert np.isclose(moves[0], 1, rtol=1e-2)
+    # next we move the time motor between zero and goal
+    assert moves[1:5] == [0, goal, 0, goal]
+    # scan should keep going
+    assert len(moves) > 20
+    # scan should not error if run
+    RE(msgs)
+
+
+@pytest.mark.timeout(PLAN_TIMEOUT)
+def test_daq_delay_scan(RE, daq, time_motor):
+    """
+    Check that daq_delay_scan's arguments all work.
+    """
+    logger.debug('test_daq_delay_scan')
+
+    msgs = list(nbp.daq_delay_scan(time_motor, [0, 1], 1, duration=0.01,
+                                   record=True))
+    configure_message = None
+    for msg in msgs:
+        if msg.command == 'configure' and msg.obj is daq:
+            configure_message = msg
+            break
+    assert configure_message is not None, 'Did not find daq configure message.'
+    assert configure_message.kwargs['record'] is True
+    assert configure_message.kwargs['controls'] == [time_motor]
+
+    # Run the scan
+    RE(msgs)
+
+
+def assert_scan_has_daq(msgs, daq):
+    """
+    Go through a plan's messages and verify it runs the daq correctly.
+
+    This is intended to be applied to any "standard" plan that is just a
+    built-in bluesky plan with the daq added to it.
+
+    We check for the following things:
+    1. The DAQ is staged (exactly once)
+    2. The DAQ is configured (exactly once)
+    3. The DAQ is triggered
+    4. The DAQ is read
+    5. The DAQ is unstaged (exactly once)
+    """
+    logger.debug('assert_scan_has_daq')
+
+    message_types = defaultdict(int)
+
+    for msg in msgs:
+        if msg.obj is daq:
+            message_types[msg.command] = message_types[msg.command] + 1
+
+    assert 'stage' in message_types, 'Scan does not stage daq.'
+    assert 'configure' in message_types, 'Scan does not configure daq.'
+    assert 'trigger' in message_types, 'Scan does not trigger daq.'
+    assert 'read' in message_types, 'Scan does not read daq.'
+    assert 'unstage' in message_types, 'Scan does not unstage daq.'
+
+    assert message_types['stage'] == 1, 'Scan stages daq multiple times.'
+    assert message_types['configure'] == 1, ('Scan configures daq multiple '
+                                             'times.')
+    assert message_types['unstage'] == 1, 'Scan unstages daq multiple times.'
```

Comment:
Also a nice pattern (message counting and usage of this function from the tests) 👍